### PR TITLE
Add devcontainer

### LINKED ISSUE #24: Add a devcontainer config file
> A bit off-topic for this PR, but would you be willing to make a PR to set up `.devcontainer/devcontainer.json`? If we, and other contributors, have a common setup for our editors, it might be easier to agree on certain styles and dev experiences. I'd be happy to offer advice if you have questions about setting up codespaces.

_Originally posted by @spenserblack in https://github.com/spenserblack/actions-wiki/pull/13#discussion_r1137861632_

### PR DESCRIPTION
**Pull request by jcbhmr**
This PR would...
- [x] Add a devcontainer configuration file
- [x] Use the base image https://github.com/devcontainers/images/tree/main/src/base-debian
- [x] Add some VS Code bash extensions by default

optional
- [ ] Add a .vscode/extensions.json file for those using vscode but not using devcontainers
- [ ] add a development note to the readme about getting started with github codespaces + devcontainers

closes #24

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "image": "mcr.microsoft.com/devcontainers/base",
+  "customizations": {
+    "vscode": {
+      "extensions": ["lizebang.bash-extension-pack"]
```
Comment (spenserblack) [inline .devcontainer/devcontainer.json:5]: Could you add the `EditorConfig.EditorConfig` extension?

**Review by spenserblack — CHANGES_REQUESTED** [review]
>Add a .vscode/extensions.json file

This generally sounds good to me. I've rarely used the desktop version of VS Code, as I'm mostly using the web version with Codespaces. Does this file auto-install the given extensions when opening the workspace?

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (jcbhmr) [reply .devcontainer/devcontainer.json]: fixed in https://github.com/spenserblack/actions-wiki/pull/36/commits/c1b57e2a290873c6b50caffc95e525d4ad703fb7

**Review by spenserblack — APPROVED** [review]
This LGTM!

Regarding me previous question, let me know if an `extensions.json` file would make sense.

IMO EditorConfig is a *required* extension (unless the contributor is very careful with their contributions :laughing:)

I haven't used the bash-extension-pack before, so I wouldn't know if it's something that's required/strongly recommended, or a nice-to-have.

**Comment by jcbhmr** [comment]
> Regarding my previous question, let me know if an `extensions.json` file would make sense.

im pretty neutral 🤷‍♂️ if you want it, add it! ❤
```json
{
  "recommended": ["lizebang.bash-extension-pack", "EditorConfig.EditorConfig"]
}
```

as an anecdote: most projects ive seen in the wild don't do it even when they have devcontainers.

the .vscode/extensions.json exists in a weird middle ground with .devcontainer/devcontainer.json existing because it's like _you're using vscode_ but yet _you're not using devcontainers_ which is a valid way to do things 🤷‍♀️ it's just a weird inbetweener stage.

**Comment by spenserblack** [comment]
Alright, let's leave it as-is :+1:

**Comment by jcbhmr** [comment]
> Does this file auto-install the given extensions when opening the workspace?

the .devcontainer/devcontainer.json will install extensions in GitHub Codespaces, but the .vscode/extensions.json WILL NOT INSTALL THEM it will just _recommend them_ to you in the bottom right with a popup to install them

lmao as i was writing this you merged it 🤣

**Pull request merged**